osm-fudge/n_gram.py

```diff
@@ -0,0 +1,15 @@
+'''N-gram function
+
+Find n-grams of a given string
+
+More: https://en.wikipedia.org/wiki/N-gram
+'''
+
+import sys
+
+def find_n_grams(n, str):
+    return [str[i:i+n] for i in range(len(str) - (n-1))]
```

Comment:
What happens if the user passes in a negative n? What about zero? Which n make sense? Can we state that as pre-condition (assert it)?

Reply:
How many n-grams are in a string of length l? Can we state that as a post-condition (assert it)?

Reply:
Couldn't get good documents for preconditon/postcondition as decorators. Did it with an assert statement

```
assert n > 0 and n <= len(str)
```

Reply:
Yep that's exactly what I meant, perfect! :bow: